index.js

```diff
@@ -63,16 +63,29 @@ module.exports = {
       // use that.
       if (typeof this._findHost === 'function') {
         target = this._findHost();
+      } else {
+        // Otherwise, we'll use this implementation borrowed from the _findHost()
+        // method in ember-cli.
+        // Keep iterating upward until we don't have a grandparent.
+        // Has to do this grandparent check because at some point we hit the project.
+        let current = this;
+        do {
+          target = current.app || app;
+        } while (current.parent.parent && (current = current.parent));
+        // If this.import is not a function, app and target should point to the same EmberApp
+        app = target;
       }
+    }
 
-      // Otherwise, we'll use this implementation borrowed from the _findHost()
-      // method in ember-cli.
-      // Keep iterating upward until we don't have a grandparent.
-      // Has to do this grandparent check because at some point we hit the project.
-      let current = this;
-      do {
-       target = current.app || app;
-      } while (current.parent.parent && (current = current.parent));
+    // We access app config and pass in the environment: 'development', 'test' or 'production'
+    // The current idea is we let user to decide situation to prefer native fetch
+    this.fetchConfig = app.project.config(app.env)['ember-fetch'] || {
+      preferNative: false
+    };
+    if (this.fetchConfig.preferNative === true) {
+      this.ui.writeInfoLine(
+        '[ember-fetch] Prefer native fetch enabled, please be aware of browser and pretender support.'
```

Comment:
I'm unsure about this logging. If we do think this logging is important, it is too vague, rather I would recommend something like:

```
[ember-fetch] fetch polyfil disabled, see: https://github.com/ember-cli/ember-fetch#something-more-details-about-this for more details
```

---

That being said, do you feel that seeing this log everytime you boot up the app is valuable? We don't do this for other polyfils, what makes this special.